Architectural-Overview.md

```diff
@@ -37,6 +37,30 @@ The language service supports the common set of a typical editor operations like
 
 * **Signature:** There are three types of signatures in the language: call, construct and index signatures.
 
+## Overview of the compilation process
+
+The process starts with preprocessing. the preprocessor figures out what files should be included in the compilation by following references (/// references and import statements).
+
+The parser then generates AST `Node`s. These are just abstract representation of the user input in a tree format. A `SourceFile` object represents an AST for a given file with some additional information like file name and source text. 
+
+The binder then passes over the AST nodes and generates and binds `Symbol`s; these are one per named entity, each `Symbol` can have multiple `Node`s, e.g.: multiple declarations of `namespace`, or `class` and a `namespace` with the same name, etc.. the binder also handles scopes and makes sure that each symbol is created in the correct enclosing function/block scope.
```

Comment:
The binder then passes over the AST nodes and generates and binds `Symbol`s.
One `Symbol` exists one per named entity.
There is a subtle distinction but several declaration nodes can name the same entity.
That means that sometimes different `Node`s will have the same `Symbol`, and each `Symbol` keeps track of its declaration `Node`s.
For example, a `class` and a `namespace` with the same name can *merge* and will have the same `Symbol`.
The binder also handles scopes and makes sure that each `Symbol` is created in the correct enclosing scope.